t/op/array.t

```diff
@@ -707,3 +707,13 @@ fresh_perl_is('my @x;$x[0] = 1;shift @x;$x[22] = 1;$x[25] = 1;','',
   {}, 'unshifting and growing an array initializes trailing elements');
 
 "We're included by lib/Tie/Array/std.t so we need to return something true";
+
+# GH #23447 - ensure that future optimizations don't break behaviour
+{
+    my @x = "a" .. "d"; sub f {} my $y = @x[2, 3, f()];
+    is $y, d, 'Non-returning final sub element in array slice in scalar context';
+
+    my @i;
+    $y = @x[2, 3, @i];
+    is $y, d, 'Empty array final element in array slice in scalar context';
```

Comment:
Why is `d` not (single- or double-)quoted here?

Reply:
Thanks, a copy & paste error or oversight. Now single quoted.